### ISSUE TITLE
Providing additional functionality to an existing module

### ISSUE THREAD

**Issue by woolite64** [nodejs/nan#657]
Let's say that I have some specialized algorithm that uses node-bignum structs and functions. I could fork this and add my own NAN_METHOD.

Is there any way to instead create a node-bignum-myalgorithm that piggybacks on the existing codebase? Or does it make more sense to maintain a fork?

**Comment by bnoordhuis** [comment]
> Is there any way to instead create a node-bignum-myalgorithm that piggybacks on the existing codebase?

I'm not sure how you envision this.  Can you provide more detail?

**Comment by woolite64** [comment]
For example, node-bignum provides:

`class BigNum : public Nan::ObjectWrap`

In theory, node-bignum-myalgorithm would have node-bignum as a dependency, then take in a BigNum as input within a NAN_METHOD.

`
NAN_METHOD(MySquareFunction)
{
  BigNum *bn = Nan::ObjectWrap::Unwrap<BigNum>(info[0]->ToObject());
  BigNum *res = new BigNum();
  BN_mul(&res->bignum_, &bn->bignum_, &bn->bignum_, ctx);

  WRAP_RESULT(res, result);
  info.GetReturnValue().Set(result);
}
`

Basically, I'm trying to separate code that I'd like to open source from custom code that I'd like to keep isolated. I'm new to NAN, so any guidance/examples would be appreciated.

**Comment by bnoordhuis** [comment]
You'd be better off asking node-bignum.  Nan neither stops nor encourages you, whether it's feasible depends on how node-bignum is implemented.

**Comment by woolite64** [comment]
Alright. Thanks :)
